src/spectral_transform.jl

```diff
@@ -434,70 +361,88 @@ function gridded!(  map::AbstractGrid{NF},                      # gridded output
+    fn = zeros(Complex{NF},nfreq_max)       # Fourier-transformed northern latitude
+    fs = zeros(Complex{NF},nfreq_max)       # Fourier-transformed southern latitude
 
     # partial sums are accumulated in alms, force zeros initially.
     fill!(alms,0)
 
     Λw = Legendre.Work(Legendre.λlm!, Λ, Legendre.Scalar(zero(NF)))
 
-    @inbounds for ilat in 1:nlat_half   # symmetry: loop over northern latitudes only
-        ilat_s = nlat - ilat + 1        # corresponding southern latitude index
+    @inbounds for ilat_n in 1:nlat_half     # symmetry: loop over northern latitudes only
+        ilat_s = nlat - ilat_n + 1          # corresponding southern latitude index
+        nlon = nlons[ilat_n]                # number of longitudes on this ring
+        nfreq  = nlon÷2 + 1                 # linear max Fourier frequency wrt to nlon
+        nfreqm = nlon÷(order+1) + 1         # (lin/quad/cub) max frequency to shorten loop over m
+        not_equator = ilat_n != ilat_s      # is the latitude ring not on equator?
 
         # Fourier transform in zonal direction
-        LinearAlgebra.mul!(fn, rfft_plan, @view(map[:,ilat]))       # Northern latitude
-        LinearAlgebra.mul!(fs, rfft_plan, @view(map[:,ilat_s]))     # Southern latitude
+        rfft_plan = rfft_plans[ilat_n]          # FFT planned wrt nlon on ring
+        js = each_index_in_ring(map,ilat_n)     # in-ring indices northern ring
+        LinearAlgebra.mul!(view(fn,1:nfreq),rfft_plan,view(map.v,js))   # Northern latitude
+
+        js = each_index_in_ring(map,ilat_s)     # in-ring indices northern ring
+                                                # Southern latitude (don't call FFT on Equator)
+                                                # then fill fs with zeros and no changes needed further down
+        not_equator ? LinearAlgebra.mul!(view(fs,1:nfreq),rfft_plan,view(map.v,js)) : fill!(fs,0)
 
         # Legendre transform in meridional direction
         # Recalculate or use precomputed Legendre polynomials Λ
         Λ_ilat = recompute_legendre ?
-            Legendre.unsafe_legendre!(Λw, Λ, lmax, mmax, cos_colat[ilat]) : Λs[ilat]
-        quadrature_weight = quadrature_weights[ilat]                # weights normalised with π/nlat
+            Legendre.unsafe_legendre!(Λw, Λ, lmax, mmax, cos_colat[ilat_n]) : Λs[ilat_n]
+        quadrature_weight = quadrature_weights[ilat_n]  # weights normalised with π/nlat
 
         lm = 1                                          # single index for spherical harmonics
-        for m in 1:mmax+1                               # Σ_{m=0}^{mmax}, but 1-based index
+        for m in 1:min(nfreq,mmax+1)                    # Σ_{m=0}^{mmax}, but 1-based index
+
+            # QUADRATURE WEIGHTS and LONGITUDE OFFSET
+            w = lon_offsets[m,ilat_n]                   # longitude offset rotation
+            # quadrature_weight *= conj(w)                # complex conjugate for back transform
```

Comment:
However, the longitude offset rotation had to be deactivated. In general we do here
```julia
    _, lons = get_colatlons(grid,nresolution)
    lon1s = [lons[each_index_in_ring(grid,i,nresolution)[1]] for i in 1:nlat]
    lon_offsets = [cispi(m*lon1/π) for m in 0:mmax, lon1 in lon1s]
```
with `lon1s` the longitude in radians of the first grid point per ring.